aten/src/ATen/core/custom_class.cpp

```diff
@@ -36,13 +36,39 @@ bool isCustomClass(const c10::IValue& v) {
       getCustomClass(v.toObject()->type()->name()->qualifiedName());
 }
 
-std::vector<std::unique_ptr<jit::Function>>& customClassMethods() {
-  static std::vector<std::unique_ptr<jit::Function>> customClassMethods;
+std::unordered_map<c10::QualifiedName, std::unique_ptr<jit::Function>>& customClassMethods() {
```

Comment:
I am skeptical of an approach for to_backend that requires changes to the fundamentals of custom class registration, so I am wary to allow functions like getCustomClassMethod.   This creates a ton of API confusion: now methods maybe are registered in a class, or maybe they are in this global map. Previously this list only existed to keep them alive and was not part of the public API for custom classes.

One major design point for our to_backend work is that it should just be helpful infrastructure for someone creating a backend. That is, it should guide the backend implementor to make good methods for serialization and object registration, but it should not be enabling something that wasn't already possible using custom classes.

Reply:
Would you be comfortable with a reintroduction of the separate registry that was present in the first version of this PR then? We cannot register these backend classes in the same way as other custom classes (and in doing so, reuse the existing class registry as-is without introducing another) because the consequent restrictions on types of custom class method arguments don't allow us to pass `IValues` back and forth between JIT and backend code (either directly or on a `Stack`) like we want to.

Reply:
Discussed offline a bit about this: the only we need a "backend registry" (or a stealth registry like this one) is to do pybinding of the `to_backend` functions, since we need to look up all the backends to bind them one at a time.

If instead we deferred the pybinding of the that function and made the user do it (which is in-line with our desire to have the `to_backend()` functions going in the backend's py module anyway), we don't need to mess with the custom class registry afaict.